packages/ui/src/hooks/use-api.ts

```diff
@@ -16,6 +18,12 @@ function useApi<Args extends any[], Response>(
   const [error, setError] = useState<RequestErrorBody>();
   const [result, setResult] = useState<Response>();
 
+  const { setLoading: setPageLoading } = useContext(PageContext);
+
+  useEffect(() => {
+    setPageLoading(loading);
+  }, [loading, setPageLoading]);
```

Comment:
This looks weird. If we have setPageLoading in the page context, why would we declare another loading state in the hook?

Reply:
good call I was thinking to keep each API's original loading status and a global loading status. But there are issues you mentioned before. Each API will overwrite the global status async. I'll simply remove it.

Reply:
np

Reply:
Updated